src/actions/market.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  type Client,
+  decodeFunctionResult,
+  encodeFunctionData,
+  parseAbi,
+} from 'viem'
+import { call } from 'viem/actions'
+import type {
+  BaseMarket,
+  MangroveActionsDefaultParams,
+} from '~mgv/types/actions/index.js'
+import { getAction } from '~mgv/utils/getAction.js'
+
+/**
+ *
+ * @param client the viem client (public or wallet)
+ * @param actionParams the parameters for the Mangrove actions
+ * @returns gets the Markets
+ */
+
+export async function getMarkets(
+  client: Client,
+  actionParams: MangroveActionsDefaultParams,
```

Comment:
You can maybe add a third argument object of type:

```ts
type TokenConfig = {
  cashness: number;
  displayDecimals?: number;
  priceDisplayDecimals?: number;
}

type GetMarketsParams = Record<string, TokenConfig>
```

Cashness is a parameter that would be the "quoteness" of a token. For example, USDT would have a high number like 1000, WETH a relatively low number like 100 and wstETH an even lower like 50. The key of the record would be the symbol of the token.

So now you can return `MarketParams[]` directly by sepcifying a base and a quote, for tokens displayDecimals and priceDisplayDecimals, you can default to the same number as in the `addresses` dir.